test/model.indexes.test.js

```diff
@@ -70,6 +70,15 @@ describe('model', function() {
       });
     });
 
+    it('should result in a clean error when calling diffIndexes() against a collection that does not exist', async () => {
+      const MySchema = new Schema({
+        name: { type: String, index: true }
+      });
+
+      const MyModel = db.model('my-model', MySchema); // This collection should not exist in DB
+      await assert.rejects(() => MyModel.diffIndexes());
```

Comment:
You should be able to remove the async function by making this `return MyModel.diffIndexes().then(() => assert.ok(false), () => {})`